artemis-commons/src/main/java/org/apache/activemq/artemis/core/server/ShutdownAwareComponent.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.core.server;
+
+/**
+ * A Component that needs to know the stop reason.
+ */
+public interface ShutdownAwareComponent extends ActiveMQComponent {
```

Comment:
I really don't like that name...

What about calling it ServiceComponent

make the method to be called stop(boolean isShutdown)


No need for these integers, and in case you want to keep a parameter, please use an enumeration.